airflow/providers/clickhouse/operators/clickhouse.py

```diff
@@ -0,0 +1,72 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import TYPE_CHECKING, Any, Callable, Dict, Optional, Sequence
+
+from airflow.models import BaseOperator
+from airflow.providers.clickhouse.hooks.clickhouse import ClickHouseHook
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class ClickHouseOperator(BaseOperator):
```

Comment:
ClickHouseQueryOperator ?

also it could maybe be base on the https://pypi.org/project/apache-airflow-providers-common-sql/ ?

Reply:
Its similar

Reply:
what is similar ?

Reply:
@pateash can you elaborate on this point?
Is there a reason why `ClickHouseHook` should not inherit from `DbApiHook`?

Reply:
HI @eladkal, @raphaelauv ,
Sorry for the late response, I was OOO.

the reason I didn't used DBApiHook is because it's intended to be used with databases ( mostly transactional supporting sqlalchemy ),

and Clickhouse being a distributed OLAP, i really think that it will be better not to expose methods like insert_rows(), which uses sqlalchemy.
rather if someone wants to have this functionality they should use the underlying library ( clickhouse-driver ) and implement their own operator using the hook and connection object.